gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -464,3 +469,179 @@ def _get_variant_calls(self, record, formats):
+                      self._file_name, data_line, str(e))
+      return MalformedVcfRecord(self._file_name, data_line, str(e))
+
+  def _convert_to_variant(self, record):
+    # type: (nucleus_proto.Variant) -> Variant
+    return Variant(
+        reference_name=record.reference_name,
+        start=record.start,
+        end=record.end,
+        reference_bases=(record.reference_bases
+                         if record.reference_bases != MISSING_FIELD_VALUE
+                         else None),
+        alternate_bases=(
+            map(str, record.alternate_bases) if record.alternate_bases else []),
+        names=map(str, record.names) if record.names else [],
+        quality=record.quality,
+        filters=(
+            [PASS_FILTER] if record.filter == [] else map(str, record.filter)),
+        info=self._get_variant_info(record),
+        calls=self._get_variant_calls(record))
+
+  def _get_variant_info(self, record):
+    info = {}
+    for k in record.info:
+      data = self._convert_list_value(record.info[k],
+                                      self._is_info_repeated.get(k, False))
+      # Prevents including missing flags as 'false' flag.
+      if isinstance(data, bool) and not data:
+        continue
+      info[k] = data
+    return info
+
+  def _convert_list_value(self, list_values, is_repeated):
+    """Converts an object of ListValue to python native types.
+
+    if is_repeated is set the output will be a list otherwise a single value.
+    """
+    output_list = []
+    for value in list_values.values:
+      if value.HasField('null_value'):
+        output_list.append(value.null_value)
+      elif value.HasField('number_value'):
+        output_list.append(value.number_value)
+      elif value.HasField('int_value'):
+        output_list.append(value.int_value)
+      elif value.HasField('string_value'):
+        output_list.append(value.string_value)
+      elif value.HasField('bool_value'):
+        output_list.append(value.bool_value)
+      elif value.HasField('struct_value'):
+        output_list.append(value.struct_value)
+      elif value.HasField('list_value'):
+        output_list.append(self._convert_list_value(value.list_value, True))
+      else:
+        raise ValueError('ListValue object has an unexpected value: %s' % value)
+
+    if is_repeated:
+      return output_list
+    if not output_list:
+      return None
+    if len(output_list) == 1:
+      return output_list[0]
+    raise ValueError('a not repeated field has more than 1 value')
+
+  def _get_variant_calls(self, record):
+    calls = []
+    for sample in record.calls:
+      call = VariantCall()
+      call.name = sample.call_set_name
+      if not sample.genotype:
+        call.genotype.append(MISSING_GENOTYPE_VALUE)
+      else:
+        for v in sample.genotype:
+          call.genotype.append(v)
+
+      phaseset_from_format = (
+          sample.info[PHASESET_FORMAT_KEY].values[0].string_value
```

Comment:
phaseset is actually an integer in the VCF spec (we use a string as we want to use a "*").
Also, should check whether it has a value using convert_list_value